### ISSUE TITLE
0.13: [Bug]: Garden fail with "Could not find build action" when referencing persistentvolumeclaim module type

### ISSUE THREAD

**Issue by eysi09** [garden-io/garden#4524]
### Current Behavior

Looks like there's a handful of issues around PVC that I'm bundling together here. 

### Reproducible example with Modules

```yaml
apiVersion: garden.io/v1
kind: Project
name: demo-project
environments:
  - name: local
    defaultNamespace: ${var.userId}
providers:
  - name: local-kubernetes
    environments: [local]
variables:
  userId: ${local.username}

---

kind: Module
name: my-pvc
type: persistentvolumeclaim
spec:
  accessMode: [ReadWriteOnce]
  resources:
    requests:
    storage: 8Gi

---

kind: Module
name: my-container
type: container
include: [backend/**/*]
dockerfile: backend/Dockerfile
services:
  - name: my-container-svc
    volumes:
      - name: my-volume
        module: my-pvc
        containerPath: /app
```

This fails with:

```
────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────
Could not find Build action my-pvc.
────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────
✖ deploy.my-container-svc → An internal error occurred while resolving status for resolve Deploy type=container name=my-container-svc (from module my-container). Here is the output:
```

And:

```
<path-to-bin>/garden-io/garden/node_modules/yoga-layout-prebuilt/yoga-layout/build/Release/nbind.js:53
        throw ex;
        ^

GraphError: Could not find Build action my-pvc.
    at MutableConfigGraph.getActionByKind (<path-to-bin>/garden-io/garden/core/src/graph/config-graph.ts:183:13)
    at MutableConfigGraph.getActionByRef (<path-to-bin>/garden-io/garden/core/src/graph/config-graph.ts:172:30)
    at <path-to-bin>/garden-io/garden/core/src/actions/base.ts:510:33
    at Array.forEach (<anonymous>)
    at BuildAction.getFullVersion (<path-to-bin>/garden-io/garden/core/src/actions/base.ts:509:23)
    at BuildAction.<anonymous> (<path-to-bin>/garden-io/garden/node_modules/typescript-memoize/src/memoize-decorator.ts:123:36)
    at BuildAction.getFullVersion (<path-to-bin>/garden-io/garden/core/src/actions/build.ts:165:33)
    at BuildAction.<anonymous> (<path-to-bin>/garden-io/garden/node_modules/typescript-memoize/src/memoize-decorator.ts:123:36)
    at BuildAction.versionString (<path-to-bin>/garden-io/garden/core/src/actions/base.ts:571:17)
    at wrapped.getInputVersion (<path-to-bin>/garden-io/garden/core/src/tasks/base.ts:240:24)
    at wrapped.complete (<path-to-bin>/garden-io/garden/core/src/graph/nodes.ts:122:31)
    at wrapped (<path-to-bin>/garden-io/garden/core/src/util/profiling.ts:154:39)
    at <path-to-bin>/garden-io/garden/core/src/graph/solver.ts:346:18 {
  detail: {
    available: {
      Build: [ 'my-container', 'backend', 'frontend' ],
      Deploy: [ 'my-pvc', 'backend', 'frontend', 'my-container-svc' ],
      Run: [ 'backend-run-task' ],
      Test: [ 'frontend-unit', 'frontend-integ' ]
    }
  },
  wrappedError: undefined,
  type: 'graph'
}

Node.js v18.14.0
```

### Reproducible example with Actions

```yaml
apiVersion: garden.io/v1
kind: Project
name: demo-project
environments:
  - name: local
    defaultNamespace: ${var.userId}
providers:
  - name: local-kubernetes
    environments: [local]
variables:
  userId: ${local.username}

---

kind: Deploy
name: my-pvc
type: persistentvolumeclaim
spec:
  spec:
    accessMode: [ReadWriteOnce]
    resources:
      requests:
      storage: 8Gi

---

kind: Build
name: my-container
type: container
include: [api/**/*]
spec:
  dockerfile: api/Dockerfile

---

kind: Deploy
name: my-container-svc
build: my-container
type: container
spec:
  volumes:
    - name: my-volume
      action: my-pvc
      containerPath: /app
```

Fails with error:

```
✖ deploy.my-container-svc → Failed processing resolve Deploy type=container name=my-container-svc (took 0.05 sec). Here is the output:

────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────
Error validating spec for Deploy type=container name=my-container-svc: Could not parse "my-pvc" as a valid action reference. An action reference should be a "<kind>.<name>" string, where <kind> is one of 'build', 'deploy', 'run' or 'test' and <name> is a valid name of an action. You may also specify an object with separate kind and name fields.. Available keys: command, args, env, cpu, memory, volumes, privileged, addCapabilities, dropCapabilities, tty, deploymentStrategy, annotations, daemon, sync, localMode, image, ingresses, healthCheck, hotReload, timeout, limits, ports, replicas)
────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────


Failed to resolve Deploy type=container name=my-container-svc: Error: resolve Deploy type=container name=my-container-svc failed: Error: Error validating spec for Deploy type=container name=my-container-svc: Could not parse "my-pvc" as a valid action reference. An action reference should be a "<kind>.<name>" string, where <kind> is one of 'build', 'deploy', 'run' or 'test' and <name> is a valid name of an action. You may also specify an object with separate kind and name fields.. Available keys: command, args, env, cpu, memory, volumes, privileged, addCapabilities, dropCapabilities, tty, deploymentStrategy, annotations, daemon, sync, localMode, image, ingresses, healthCheck, hotReload, timeout, limits, ports, replicas)
```

### Expected behavior

Expecting both to run without validation errors. 

### Workaround

Stay on 0.12 until this is fixed.

`garden version` 0.13.0

**Comment by eysi09** [comment]
Also, the `persistentvolumeclaim` Deploy action has nested specs, is that intentional?

https://docs.garden.io/reference/action-types/configmap/persistentvolumeclaim#spec.spec

**Comment by vvagaytsev** [comment]
@eysi09 I think so. It used to be a root-level `spec` field in `0.12`. Now, each action has a kind-specific `spec` field on the root level, so this spec became a nested one.

**Comment by vvagaytsev** [comment]
@Orzelius could this also be addressed in #4523?

**Comment by eysi09** [comment]
> @eysi09 I think so. It used to be a root-level `spec` field in `0.12`. Now, each action has a kind-specific `spec` field on the root level, so this spec became a nested one.

It's a pretty awkward syntax and we should fix in 0.14 (or sooner and deprecate `spec.spec` and remove with 0.14).

**Comment by Orzelius** [comment]
I kinda like it, as it has all the kubernetes specific spec there. If we were to add our own spec fields they'd be separate and easy to distinguish
